test/integration/client-side-encryption/client_side_encryption.prose.test.js

```diff
@@ -1723,6 +1723,120 @@ describe('Client Side Encryption Prose Tests', metadata, function () {
     });
   });
 
+  context('13. Unique Index on keyAltNames', function () {
+    let client, clientEncryption, setupKeyId;
+
+    beforeEach(async function () {
+      // Create a MongoClient object (referred to as client).
+      client = this.configuration.newClient();
+      await client.connect();
+
+      // Using client, drop the collection keyvault.datakeys.
+      await client
+        .db('keyvault')
+        .dropCollection('datakeys')
+        .catch(() => null);
+
+      await client
+        .db('keyvault')
+        .collection('datakeys')
+        .createIndex(
+          { keyAltNames: 1 },
+          {
+            unique: true,
+            partialFilterExpression: { keyAltNames: { $exists: true } },
+            writeConcern: { w: 'majority' }
+          }
+        );
+
+      // Create a ClientEncryption object (referred to as client_encryption) with client set as the keyVaultClient.
+      clientEncryption = new this.configuration.mongodbClientEncryption.ClientEncryption(client, {
+        keyVaultNamespace: 'keyvault.datakeys',
+        kmsProviders: getKmsProviders()
+      });
+
+      // Using client_encryption, create a data key with a local KMS provider and the keyAltName "def".
+      setupKeyId = await clientEncryption.createDataKey('local', {
+        keyAltNames: ['def']
+      });
+    });
+
+    afterEach(async () => {
+      clientEncryption = null;
+      setupKeyId = null;
+      await client.close();
+    });
+
+    context('Case 1', metadata, function () {
+      it('createDataKey() handles duplicate key errors on the keyvault collection', async function () {
+        // 1. Use client_encryption to create a new local data key with a keyAltName "abc" and assert the operation does not fail.
+        await clientEncryption.createDataKey('local', {
+          keyAltNames: ['abc']
+        });
+
+        // 2. Repeat Step 1 and assert the operation fails due to a duplicate key server error (error code 11000).
+        const maybeErrorStep2 = await clientEncryption
```

Comment:
what do you think of using `resultStep2` instead of `maybeError` here and throughout this set of tests? I am just thinking that naming variables what we expect to receive in the test would make it a bit easier to follow the logic, since it's not that it may or may not be an error and either is acceptable for this test, it's that it *should* be an error result; and then further down, we *expect* the result to *not* be an error.